### PR TITLE
bug/anagramica_fixes

### PR DESCRIPTION
**Pull request by jonhindle**
### Context

> Does this issue have a Trello card?
https://trello.com/c/PsTYcXN7/689-anagramica-sentry-errors

> If this is an issue, do we have steps to reproduce?
https://sentry.io/organizations/ministryofjustice/issues/3021722312/?project=5377056&query=is%3Aunresolved

### Intent

> What changes are introduced by this PR that correspond to the above card?
Fixes to post route.

> Would this PR benefit from screenshots?
n/a

### Considerations

> Is there any additional information that would help when reviewing this PR?
n/a

> Are there any steps required when merging/deploying this PR?
n/a

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

**Review by davidthomas22 — APPROVED** [review]
LGTM

**Review by andrewrlee — DISMISSED** [review]
LGTM

File: server/routes/createAnagramicaRouter.js [hunk 1]
```diff
@@ -22,11 +22,16 @@ const createAnagramicaRouter = config => {
...
-        const word = toAlpha(rawWord);
-        return Object.assign(total, { [word]: finder.find(word) });
-      }, {});
+      const scores =
+        Array.isArray(words) && words.length > 0
+          ? words.reduce((total, rawWord) => {
+              const word = toAlpha(rawWord);
+              return Object.assign(total, { [word]: finder.find(word) });
+            }, {})
+          : [];
```
Comment (andrewrlee) [inline server/routes/createAnagramicaRouter.js:34]: it's weird you return an empty array in on case and an object in another, should this default to an empty object?

(you presumably don't need the length check as reducing an empty array is a noop)

Comment (jonhindle) [reply server/routes/createAnagramicaRouter.js:34]: good catch, should be an object 😞

File: server/routes/createAnagramicaRouter.js [hunk 2]
```diff
@@ -22,12 +23,19 @@ const createAnagramicaRouter = config => {
     )
     .post((req, res) => {
       const { letters, words = [] } = req.body;
+      if (letters.length !== 10 && !/^[a-z]/.test(letters)) {
```
[the remark this answers is not in the document]
Comment (jonhindle) [reply server/routes/createAnagramicaRouter.js:26]: Yeah the game always provides ten letters.

File: server/routes/createAnagramicaRouter.js [hunk 3]
```diff
@@ -22,12 +23,19 @@ const createAnagramicaRouter = config => {
...
+        return res.send({ error: 'Invalid letter selection' });
+      }
       const best = finder.best(toAlpha(letters));
-      const scores = words.reduce((total, rawWord) => {
-        const word = toAlpha(rawWord);
-        return Object.assign(total, { [word]: finder.find(word) });
-      }, {});
-      res.send({
+      const scores =
+        Array.isArray(words) && words.length > 0
```
Comment (andrewrlee) [inline server/routes/createAnagramicaRouter.js:32]: meh. don't need length check..

**Review by andrewrlee — APPROVED** [review]
LGTM

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**